docs/DeveloperGuide.md

```diff
@@ -1,29 +1,115 @@
+[DeveloperGuide.md](DeveloperGuide.md)
 # Developer Guide
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Command Mechanisms:
+For all valid commands, the mechanism of implementation are as follows:
+1. Read input - ```runBagPacker()``` method in ```BagPacker``` calls the ```Parser``` class to read user input command
+2. Create command object - The ```Parser``` class creates a corresponding command object of the relevant command
+3. Execute command object - ```runBagPacker()``` method executes the ```.execute()``` method (overridden by child classes) of the command object 
+   which runs the actual command function
+
+#### Add Command
+
+Add command is used to add a quantity of item(s) to the packing list.
+
+Mechanism: ```AddCommand.execute()``` calls the ```PackingList.addItem()``` method from the ```PackingList``` class which executes the ```ArrayList.add()``` method to add the item to the ```PackingList``` ArrayList. 
+It then updates the ```quantity``` variable according to the quantity inputted by the user.
+
+
+#### Preventing duplicate items
+
+When using the `add` function, we have decided to implement a passive function that checks whether the item with the same name already exists in the packingList.
+
+This is done through the `contains()` method in class `PackingList()`, which is called during `execute` in an `AddCommand` object.
+
+Below is the UML diagram showing what occurs during `add` function.
+![img_1.png](img_1.png)
```

Comment:
<img width="956" alt="Screenshot 2023-03-30 at 9 10 14 AM" src="https://user-images.githubusercontent.com/50757300/228702962-b7fb3119-cbeb-494f-ac70-eee741358f40.png">

Should the calls to ui be pointed at its lifeline rather than the entity, and be aligned to its activation bar? And, shouldn't the alt block contain the two possibilities of calls to the UI entity?